test/semantics/test_any.sh

```diff
@@ -48,9 +48,9 @@ function internal_check() {
 }
 
 gr=0
-for input in ${srcdir}/$*; do
+for input in "$@"; do
```

Comment:
Found it...GitHub hiding things.

test_any.sh is called the way you said it's called now, so nothing has changed (except the paths work from lit's perspective instead of the script's). I copied this bit from test_errors.sh, trying to make them all run the same way (due to the issue of paths getting mixed up)

```
foreach(test ${LABEL_TESTS} ${CANONDO_TESTS} ${DOCONCURRENT_TESTS}
             ${CRITICAL_TESTS} ${GETSYMBOLS_TESTS} ${GETDEFINITION_TESTS})
  add_test(NAME ${test}
    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/test_any.sh ${test} ${F18})
endforeach()
```

Reply:
Apologies - I misunderstood the cmake syntax there. Now you have explained I can see that the lists are concatenated and expanded so test takes the each value in this iteration and the multiple file examples are actually globs in the lists themselves.

So I guess what we are seeing here is a mostly harmless bug in the original script in that it will try and treat the compiler passed on the last argument as a test input. When I run a test I see this: 
`$ ./test_any.sh critical04.f90 /home/ricbar01/ossf18/Release/bin/flang

PASS  ./critical04.f90
PASS  /home/ricbar01/ossf18/Release/bin/flang` 
Somehow $input on that second iteration of the loop is the absolute path and not $srcdir/<path> so it passes the check on filename with that absolute path. I guess that is a neat bash trick.

In this case, I think your changes to preserve the original semantics are fine.